eox_nelp/external_certificates/models.py

```diff
@@ -0,0 +1,102 @@
+"""
+Models file for the external_certificates.
+"""
+import logging
+
+from django.contrib.auth import get_user_model
+from django.db import models
+from django.utils import timezone
+
+User = get_user_model()
+logger = logging.getLogger(__name__)
+
+
+class ExternalCertificate(models.Model):
+    """
+    Model to store information about external certificates.
+
+    Attributes:
+        user (User): The user associated with the certificate.
+        group_code (str): Group code associated to the cert.
+        certificate_id (str): Unique identifier of the certificate.
+        course_id (str): Group code used to generate the certificate.
+        certificate_url_en (str): URL to access the certificate in English.
+        certificate_url_ar (str): URL to access the certificate in Arabic.
+        created_at (datetime): Timestamp when the certificate was created.
+    """
+    user = models.ForeignKey(
+        User,
+        on_delete=models.CASCADE,
+        related_name="external_certificates",
+        help_text="The user associated with this certificate."
+    )
+    group_code = models.CharField(
+        max_length=50,
+        help_text="Group code related the cert. Normally mapped to course_id"
+    )
+    certificate_id = models.CharField(
+        max_length=50,
+        help_text="Unique identifier for the certificate."
+    )
+    certificate_url_en = models.URLField(
+        max_length=500,
+        help_text="URL for the certificate in English."
+    )
+    certificate_url_ar = models.URLField(
+        max_length=500,
+        help_text="URL for the certificate in Arabic."
+    )
+    created_at = models.DateTimeField(
+        default=timezone.now,
+        help_text="The date and time when the certificate was created."
+    )
+
+    class Meta:  # pylint: disable=too-few-public-methods
+        """
+        Meta class for ExternalCertificate model.
+        """
+        verbose_name = "External Certificate"
+        verbose_name_plural = "External Certificates"
+
+    def __str__(self):
+        return f"Certificate {self.certificate_id} for user {self.user}"
+
+    @classmethod
+    def create_external_certificate_from_certificate_response(cls, certificate_response, user):
+        """Create an external certificate from the response of the external certificate service.
+        Args:
+            certificate_response (dict): The response from the external certificate service.
+            user (User): The user associated with the certificate.
+        Logs:
+            Logs the success or failure of the certificate creation
+        Returns:
+            ExternalCertificate: The created external certificate instance or None
+        """
+        if certificate_response.get("error"):
```

Comment:
I haven't seen the implementation yet but how this is possible I mean why you don't validate this before calling this method

Reply:
Because in the Lamdas I have a flag to decide the bypass of an error from the external certificate service. And continue... This is most for staging purposes an guarantee that the external_certificate_respose would have a good shape.

```python3
    ## Check if the certificate was created successfully
    if certificate_response.get("error") and int(os.environ.get("VALIDATE_EXTERNAL_CERTIFICATES_RESPONSE_ERROR", 1)):
        raise ValueError(f"Failed to create certificate for national_id {certificate_data_event.national_id}")
```